Documentation/Browser/BrowsingContextsAndNavigables.md

```diff
@@ -0,0 +1,128 @@
+be some REPL specific global object that has global functions to call for doing things like loading
+from the filesystem, or even be as complex as Node or Bun. In a Browser context, the Global object is
+normally the Window, unless there's a Worker involved. This concept is quite different from a native
+executable, where there's no actual object representing the global scope, simply symbols that the
+linker and loader make available to each module.
+
+While the Global Object and its Global Environment represent the root of the tree of identifiers visible
+to the executing JavaScript code, the Global Object isn't sufficient to model all the state around
+a conceptual thread of execution in ECMAScript. This is where the two concepts of [*Realms*](https://tc39.es/ecma262/#sec-code-realms)
+and [*Execution Contexts*](https://tc39.es/ecma262/#sec-execution-contexts) come into play.
+A [*Realm Record*](https://tc39.es/ecma262/#realm-record) is a container that holds a global object,
+its associated Global Environment, a set of intrinsic objects, and any *host* (also called an *embedder*
+in some specification documents) defined extra state that needs to be associated with the realm.
+In LibWeb, the Host Defined slot holds an object that has the HTML Environment Settings Object for each realm,
+as well as all the prototypes, constructors, and namespaces that need to be exposed on the Global Object
+for Web APIs. On top of the Realm abstraction, ECMAScript uses the Execution Context to model the state
+of execution of one particular script or module. On top of the Realm it refers to, the Execution Context
+has information about the current function, the script or module that the current code block belongs to,
+additional Environment Records required to access names in the current scope, any running generator state,
+and most importantly to the thread analogy, the state needed to suspend and resume execution of that script.
+Throughout the ECMAScript specification, the text refers to a stack of execution contexts, with the uppermost
+one always being the [*running execution context*](https://tc39.es/ecma262/#running-execution-context).
+As with Environment Records, Realms and Execution contexts are not visible to running JavaScript code.
+
+The final missing piece for the JavaScript execution model is how these stacks of Execution Contexts
+are actually scheduled to run by the ECMAScript implementation. In the most common case, this means directly
+mapping the ECMAScript model to the earlier native concepts of threads and processes in a way that
+allows for flexibility in the implementation strategies. The last thing that the specification authors want
+to do is constrain implementations so much that innovation and experimentation becomes impossible.
+The method for this mapping is the two related specification objects [*Agents*](https://tc39.es/ecma262/#sec-agents)
```

Comment:
```suggestion
The method for this mapping is the two related specification mechanisms [*Agents*](https://tc39.es/ecma262/#sec-agents)
```

(wording from the spec: `An agent is a specification mechanism and need not correspond to any particular artefact of an ECMAScript implementation.`)